mega-zep-backend/src/main/java/com/gepardec/mega/service/impl/monthlyreport/MonthlyReportServiceImpl.java

```diff
@@ -40,13 +43,22 @@ public MonthlyReport getMonthendReportForUser(final String userId) {
 
     @Override
     public boolean setOpenAndUnassignedStepEntriesDone(Employee employee, Long stepId) {
-        return stepEntryService.setOpenAndAssignedStepEntriesDone(employee, stepId);
+        LocalDate from = LocalDate.parse(DateUtils.getFirstDayOfFollowingMonth(employee.releaseDate()));
+        LocalDate to = LocalDate.parse(DateUtils.getLastDayOfFollowingMonth(employee.releaseDate()));
+
+        return stepEntryService.setOpenAndAssignedStepEntriesDone(employee, stepId, from, to);
     }
 
     private MonthlyReport calcWarnings(List<ProjectEntry> projectEntries, Employee employee) {
         final List<JourneyWarning> journeyWarnings = warningCalculator.determineJourneyWarnings(projectEntries);
         final List<TimeWarning> timeWarnings = warningCalculator.determineTimeWarnings(projectEntries);
-        final List<Comment> comments = commentService.findCommentsForEmployee(employee);
+
+        final List<Comment> comments = new ArrayList<>();
+        if(employee != null) {
+            LocalDate fromDate = LocalDate.parse(DateUtils.getFirstDayOfFollowingMonth(employee.releaseDate()));
+            LocalDate toDate = LocalDate.parse(DateUtils.getLastDayOfFollowingMonth(employee.releaseDate()));
+            comments.addAll(commentService.findCommentsForEmployee(employee, fromDate, toDate));
```

Comment:
@gattma do we get here only comments of the current month? We need to get all open comments (even from past months) and all closed from the current month